### PR TITLE
add oauth credentials to MultiProviderAuthDialog

### PR DESCRIPTION
**Pull request by lzc850612**
### Purpose
*What is the context of this pull request? Why is it being done?*

Add OAuthCredentials to MultiProviderAuthDialog to support using a different credentials for auth
use LG for MultiProviderAuthDialog
Remove ResponseManager and related classes

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*

### Tests
*Is this covered by existing tests or new ones? If no, why not?*

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*

#### General
- [ ] I have commented my code, particularly in hard-to-understand areas	
- [ ] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

### CODE REVIEW

**Comment by xieofxie** [comment]
Could we use lg in solution library now? since I think lg files will not be published to nuget?

File: sdk/csharp/libraries/microsoft.bot.solutions/Testing/BotTestBase.cs [hunk 1]
```diff
@@ -1,60 +1,12 @@
...
 // Licensed under the MIT License.
 
-using System.Collections.Specialized;
-using System.Text.RegularExpressions;
-using Microsoft.Bot.Solutions.Responses;
+using System;
 
 namespace Microsoft.Bot.Solutions.Testing
 {
+    [Obsolete("This class is being deprecated. Please use Microsoft.Bot.Solutions.Responses.LocaleTemplateManager to handle response generation.")]
```
Comment (ryanisgrig) [inline sdk/csharp/libraries/microsoft.bot.solutions/Testing/BotTestBase.cs:11]: Is this the right message?

Comment (ryanisgrig) [reply sdk/csharp/libraries/microsoft.bot.solutions/Testing/BotTestBase.cs:11]: Nevermind, I see it was only used for ResponseManager

Comment (lzc850612) [reply sdk/csharp/libraries/microsoft.bot.solutions/Testing/BotTestBase.cs:11]: yeah a bit confusing but the functions are all for response generation

**Comment by lzc850612** [comment]
> Could we use lg in solution library now? since I think lg files will not be published to nuget?

Developers should be able to use MultiProviderAuthDialog without having to passing a whole set of LG responses. That means we need to include the LG files. We should be able to do it by including them explicitly in the .nuspec file. I'm having trouble making the nuspec work now with azure build pipeline so still working on it (getting @darrenj to help)

**Comment by xieofxie** [comment]
> > Could we use lg in solution library now? since I think lg files will not be published to nuget?
> 
> Developers should be able to use MultiProviderAuthDialog without having to passing a whole set of LG responses. That means we need to include the LG files. We should be able to do it by including them explicitly in the .nuspec file. I'm having trouble making the nuspec work now with azure build pipeline so still working on it (getting @darrenj to help)

Good to know the nuspec function! I once thought we could load content from assembly but it seems SDK doesn't expose ParseText in MultiLanguageLG

**Review by lauren-mills — APPROVED** [review]

**Comment by lzc850612** [comment]
Sadly even if we did include the LG files in the nuget files, they still don't get copied over to the project. So i'm reverting back to use ResponseManager. Please take a look.

**Pull request merged**